XMonad/Layout/Inspect.hs

```diff
@@ -69,7 +109,12 @@ cast' x | Just HRefl <- ta `eqTypeRep` tb = x
 
 type family InspectResult i
 
--- | TODO: for layout/modifier authors
+-- | A typeclass defining 'inspectLayout' over monoidal @InspectResult@ types.
+-- An overlappable instance provides sensible default behavior, returning
+-- 'mempty'. (Specific @InspectResult@ instances may override this by defining
+-- their own instance of 'InspectLayout'.) Additionally, an instance is provided
+-- for layouts using 'Choose', provided that all the layouts in the 'Choose' are
+-- capable of being inspected.
```

Comment:
These should perhaps be documentation of the respective instances?  Although I guess the `Choose` one could be omitted entirely, as the dependency graph is right there in the type signature